p4/include/tps_checksum.p4

```diff
@@ -19,29 +19,69 @@
 *************************************************************************/
 
 control TpsVerifyChecksum(inout headers hdr, inout metadata meta) {
-    apply {  }
+    apply {
+        verify_checksum(hdr.ipv4.isValid(),
+            {
+	        hdr.ipv4.version,
+	        hdr.ipv4.ihl,
+                hdr.ipv4.diffserv,
+                hdr.ipv4.totalLen,
+                hdr.ipv4.identification,
+                hdr.ipv4.flags,
+                hdr.ipv4.fragOffset,
+                hdr.ipv4.ttl,
+                hdr.ipv4.protocol,
+                hdr.ipv4.srcAddr,
+                hdr.ipv4.dstAddr
+            },
+            hdr.ipv4.hdrChecksum,
+            HashAlgorithm.csum16
+        );
+
+	verify_checksum(hdr.udp_int.isValid(),
+            {
+                hdr.udp_int.src_port,
+                hdr.udp_int.dst_port,
+                hdr.udp_int.len
+            },
+            hdr.udp_int.cksum,
+            HashAlgorithm.csum16
+        );
+    }
 }
 
 /*************************************************************************
 *************   C H E C K S U M    C O M P U T A T I O N   **************
 *************************************************************************/
 
 control TpsComputeChecksum(inout headers  hdr, inout metadata meta) {
-     apply {
-	update_checksum(
-	    hdr.ipv4.isValid(),
-            { hdr.ipv4.version,
-	      hdr.ipv4.ihl,
-              hdr.ipv4.diffserv,
-              hdr.ipv4.totalLen,
-              hdr.ipv4.identification,
-              hdr.ipv4.flags,
-              hdr.ipv4.fragOffset,
-              hdr.ipv4.ttl,
-              hdr.ipv4.protocol,
-              hdr.ipv4.srcAddr,
-              hdr.ipv4.dstAddr },
+    apply {
+        update_checksum(hdr.ipv4.isValid(),
+            {
+	        hdr.ipv4.version,
+	        hdr.ipv4.ihl,
+                hdr.ipv4.diffserv,
+                hdr.ipv4.totalLen,
+                hdr.ipv4.identification,
+                hdr.ipv4.flags,
+                hdr.ipv4.fragOffset,
+                hdr.ipv4.ttl,
+                hdr.ipv4.protocol,
+                hdr.ipv4.srcAddr,
+                hdr.ipv4.dstAddr
```

Comment:
Indentation is off.

Could this be done with a struct or something to eliminate all of the duplicate lines?

Reply:
Didn't notice them earlier but I've already have that fix with the telem report work I'm doing.